pkg/types/v1/fs.go

```diff
@@ -22,19 +22,22 @@ import (
 )
 
 type FS interface {
-	Open(name string) (*os.File, error)
-	Chmod(name string, mode os.FileMode) error
+	Chmod(name string, mode fs.FileMode) error
 	Create(name string) (*os.File, error)
-	Mkdir(name string, perm os.FileMode) error
-	Stat(name string) (os.FileInfo, error)
-	Lstat(name string) (os.FileInfo, error)
-	RemoveAll(path string) error
+	Glob(pattern string) ([]string, error)
+	Link(oldname, newname string) error
```

Comment:
This `Link` support for hardlinks the main reason for this change. V1 did not include support for hardlinks.